src/components/input/SelectNext.tsx

```diff
@@ -60,7 +60,35 @@ function SelectOption<T>({
   }
 
   const { selectValue, value: currentValue } = selectContext;
-  const selected = !disabled && currentValue === value;
+  const valueIsArray = Array.isArray(currentValue);
```

Comment:
I think the multi vs single selection behavior should be provided explicitly by the context rather than trying to infer it from the type of `value`. Otherwise the component could behave in an unexpected way if you tried to use single-selection with items that were tuples.

Reply:
True. We need to expose `multiple` via context as well.

I suspect TypeScript is going to force us to still verify `currentValue` is an array so that we can use it as such. Maybe I can try what you attempted [here](https://github.com/hypothesis/frontend-shared/pull/1601#issuecomment-2205588786), but only in the context type.